Module not found: Error: Can't resolve 'graphql'

### Prerequisites

- [X] I confirm my issue is not in the [opened issues](https://github.com/mswjs/msw/issues)
- [X] I confirm the [Frequently Asked Questions](https://mswjs.io/docs/faq) didn't contain the answer to my issue

### Environment check

- [X] I'm using the [latest](https://github.com/mswjs/msw/releases/latest) `msw` version
- [X] I'm using Node.js version 18 or higher

### Browsers

Chromium (Chrome, Brave, etc.)

### Reproduction repository

https://github.com/kguptaping/msw-issue

### Reproduction steps

`cd <folder>`
`yarn`
`npx msw init public --save`
`yarn start`

### Current behavior

Error: 
`ERROR in ./node_modules/msw/lib/core/utils/internal/parseGraphQLRequest.mjs 17:12-29
Module not found: Error: Can't resolve 'graphql' in '/Users/<user>/codebase/node_modules/msw/lib/core/utils/internal'
` 

### Expected behavior

Should not show any issue.

--- a/package.json
+++ b/package.json
@@ -142,6 +142,7 @@
     "@types/cookie": "^0.6.0",
     "@types/statuses": "^2.0.4",
     "chalk": "^4.1.2",
+    "graphql": "^16.8.1",
     "headers-polyfill": "^4.0.2",
     "is-node-process": "^1.2.0",
     "outvariant": "^1.4.2",
@@ -180,7 +181,6 @@
     "fs-extra": "^11.2.0",
     "fs-teardown": "^0.3.0",
     "glob": "^10.3.10",
-    "graphql": "^16.8.1",
     "jsdom": "^23.2.0",
     "json-bigint": "^1.0.0",
     "lint-staged": "^15.2.0",
@@ -200,13 +200,9 @@
     "webpack-http-server": "^0.5.0"
   },
   "peerDependencies": {
-    "graphql": ">= 16.8.x",
     "typescript": ">= 4.8.x"
   },
   "peerDependenciesMeta": {
-    "graphql": {
-      "optional": true
-    },
     "typescript": {
       "optional": true
     }

